### PR TITLE
lakectl local: add init esti tests

### LINKED ISSUE #6375: lakectl local: Esti tests for init

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #6375

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]

File: cmd/lakectl/cmd/diff.go [hunk 1]
```diff
@@ -109,13 +110,16 @@ func printDiffBranch(ctx context.Context, client api.ClientWithResponsesInterfac
...
 func printDiffRefs(ctx context.Context, client api.ClientWithResponsesInterface, left, right *uri.URI, twoDot bool) {
 	diffs := make(chan api.Diff, maxDiffPageSize)
-	err := diff.StreamRepositoryDiffs(ctx, client, left, right, "", diffs, twoDot)
-	if err != nil {
-		DieErr(err)
-	}
+	var wg errgroup.Group
+	wg.Go(func() error {
+		return diff.StreamRepositoryDiffs(ctx, client, left, right, "", diffs, twoDot)
+	})
```
Comment (nopcoder) [inline cmd/lakectl/cmd/diff.go:116]: If `diff.StreamRepositoryDiffs` will create/close the channel, process the diff using goroutine inside the func - the use of this call will not require all the callers to `go` each time we use it.

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: cmd/lakectl/cmd/local_init.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ func localInit(ctx context.Context, dir string, remote *uri.URI, force bool) (*l
 	ignoreFile, err := git.Ignore(dir, []string{dir}, []string{filepath.Join(dir, local.IndexFileName)}, local.IgnoreMarker)
 	if err == nil {
 		fmt.Println("location added to", ignoreFile)
-	} else if !errors.Is(err, git.ErrNotARepository) {
+	} else if !(errors.Is(err, git.ErrNotARepository) || errors.Is(err, git.ErrNoGit)) {
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/local_init.go:44]: Agree, currently handled in another PR

File: cmd/lakectl/cmd/diff.go (hunk 1 above)
Comment (N-o-Z) [reply cmd/lakectl/cmd/diff.go:116]: Deferring this comment to another issue, as we needed to also be able to handle errors that resulted from this function call.

File: esti/lakectl_local_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/local"
+)
+
+func listDir(t *testing.T, dir string) []string {
+	var files []string
+	err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
```
Comment (nopcoder) [inline esti/lakectl_local_test.go:16]: Check `err` and return if not nil before checking `d.IsDir`

**Review by nopcoder — APPROVED** [review]
Approved with minor comment on err check

**Pull request merged**